cmd/goal/clerk.go

```diff
@@ -754,6 +754,9 @@ var compileCmd = &cobra.Command{
 				// Check if from was specified, else use default
 				if account == "" {
 					account = accountList.getDefaultAccount()
+					if account == "" {
+						reportErrorln("cannot find a default account")
```

Comment:
better error message? "no default account set. set one with `goal ...` or specify an account with -a" (and fill in what `goal ...` might be to set a default account.

Reply:
better than nothing :) 
addressed.